packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -213,6 +213,79 @@ enum BottomNavigationBarType {
 /// }
 /// ```
 /// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material}
+/// This example shows [BottomNavigationBar] used in a [Scaffold] Widget with
+/// different interaction patterns. Tapping on the first [BottomNavigationBarItem]
+/// uses the [ScrollController] to animate the [ListView] to the top. The second
+/// [BottomNavigationBarItem] opens a [DropdownButton].
+///
+///```dart
+/// int _selectedIndex = 1;
+/// final ScrollController _homeController = ScrollController();
+///
+/// void _onItemTapped(int index) {
+///   if (index == 0) {
+///     _homeController.animateTo(
+///        0.0,
+///        duration: const Duration(milliseconds: 500),
+///        curve: Curves.easeOut,
+///     );
+///   } else {
+///     setState(() {
+///       _selectedIndex = index;
+///     });
+///   }
+/// }
+///
+/// Widget _listViewBody() {
+///   return ListView.separated(
+///       controller: _homeController,
+///       itemBuilder: (BuildContext context, int index) {
+///         return Center(
+///             child: Text(
+///               'Item $index',
+///               style: TextStyle(
+///                 fontWeight: _selectedIndex == 1? FontWeight.bold : FontWeight.normal,
```

Comment:
This line could directly leverage a `FontWeight` attribute if the tap handler stored one of those instead of an integer.